dependencies/README.md

```diff
@@ -88,3 +90,90 @@ The tool allows users to choose the scope of entry point classes:
   - **All classes in the input class path**: when reachability check is off, then
     all static linkage errors from all classes in the classpath, regardless of the reachability,
     are reported.
+
+<a name="maven_dependency_graph"></a>
+# Maven Dependency Graph
+
+A Maven dependency graph is a graph data structure where
+- Node: a node is a Maven artifact identified by [Maven coordinates][1] such as
+  `com.example.foo:bar:1.5.4`. A node may be marked _unavailable_ when the artifact is not available
+  in Maven repositories.
+- Edge: a directed edge is a dependency from one Maven artifact (_source_ of the edge)
+  to another Maven artifact (_target_ of the edge).
+
+  A dependency has a boolean attribute `optional` and an enum attribute `scope`,
+  among other properties listed in [POM Reference: Dependencies][2].
+
+  Self-loops are not possible. A parallel edge is allowed but is dropped in the model.
+
+## Comparison with Maven Dependency Trees by pom.xml
+
+Compared to a _Maven dependency tree_ retrieved by a pom.xml through
+`RepositorySystem.resolveDependencies`, a Maven dependency graph has the following differences:
+
+- A Maven dependency tree cannot have multiple versions of a Maven artifact, whereas a Maven
+  dependency graph can have them.
+- A Maven dependency tree does not have transitive `scope: provided` dependencies under
+  a `scope: provided` dependency, whereas a Maven dependency graph can have them.
+- A Maven dependency tree does not contain a cycle,
```

Comment:
Really, how?

Reply:
A Maven dependency tree (post mediation) does not contain a cycle.

For example, a cycle in a dependency graph is dom4j and jaxen:
[dom4j:dom4j:jar:1.5.2](https://search.maven.org/artifact/dom4j/dom4j/1.5.2/jar) depends on [jaxen:jaxen:jar:1.1-beta-4](https://search.maven.org/artifact/jaxen/jaxen/1.1-beta-4/jar), which depends on dom4j:dom4j:jar:1.5.2. This is a cycle in a graph.

However, the Maven dependency tree (post-mediation) does not show a cycle, because such node is omitted.

```
$ mvn dependency:tree -Dverbose
[INFO] com.google.suztomo:cycle:jar:1.0-SNAPSHOT
[INFO] \- dom4j:dom4j:jar:1.5.2:compile
[INFO]    +- jaxme:jaxme-api:jar:0.3:compile
[INFO]    +- (dom4j:dom4j:jar:1.5.2:compile - omitted for cycle)
[INFO]    +- jaxen:jaxen:jar:1.1-beta-4:compile
[INFO]    +- msv:xsdlib:jar:20030807:compile
[INFO]    +- msv:relaxngDatatype:jar:20030807:compile
[INFO]    +- pull-parser:pull-parser:jar:2:compile
[INFO]    +- xpp3:xpp3:jar:1.1.3.3:compile
[INFO]    +- stax:stax-api:jar:1.0:compile
[INFO]    \- xml-apis:xml-apis:jar:1.0.b2:compile
```
(`dom4j:dom4j` appears only once)